src/database/hooks/useLocalDatabaseHook.js

```diff
@@ -36,7 +37,17 @@ const useLocalDatabaseHook = (withMigration = false) => {
     }));
   };
 
-  const {channelInfo, channelList, channelMember, chat, user} = databaseListener;
+  const debouncedRefresh = React.useCallback(
+    _.debounce((key) => refresh(key)),
+    50
+  );
+
+  const refreshWithId = (key, id) => {
+    const keyWithId = `${key}_${id}`;
+    debouncedRefresh(keyWithId);
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `debouncedRefresh` function is created using `_.debounce`, which is good for performance as it prevents unnecessary refreshes when called in quick succession. However, the `refreshWithId` function creates a new key by concatenating the provided key and id with an underscore. This could potentially lead to collisions if keys or ids contain underscores. Consider using a more unique separator or a different method to combine the key and id.



<!-- This is an auto-generated comment by OSS CodeRabbit -->